### PR TITLE
approx1+2 port to oneapi. tests out aside from Subs, JIT, Memory

### PR DESCRIPTION
**Pull request by pv-pterab-s**
Description
-----------
These changes constitute a partially-functioning OneAPI port of the "approx1" and "approx2" functions. Its tests pass except for those relying on unsupported ArrayFire internals:

 - `*TypeParam*Double*` : ArrayFire erroneously reports double's not supported.
 - `*SubArray*` : ArrayFire internal indexing not yet complete.
 - `*Batch*` : ArrayFire unsupported JIT.
 - `*MaxDims*` : ArrayFire max-dims definition not yet in parity with OneAPI.

 - `approx1` passes with gtest: 

```
./test/approx1_oneapi --gtest_filter="-Approx1.CPPNearestBatch:Approx1.CPPLinearBatch:Approx1.CPPCubicBatch:Approx1.CPPNearestMaxDims:Approx1.CPPLinearMaxDims:Approx1.CPPCubicMaxDims:Approx1.OtherDimLinear:Approx1.OtherDimCubic:Approx1.CPPInfCheck:Approx1.CPPUniformInfCheck:Approx1V2Simple/1.UseNullOutputArray:Approx1V2Simple/1.UseFullExistingOutputArray:Approx1V2Simple/1.UseExistingOutputSubArray:Approx1V2Simple/1.UseReorderedOutputArray:Approx1V2Simple/1.UniformUseNullOutputArray:Approx1V2Simple/1.UniformUseFullExistingOutputArray:Approx1V2Simple/1.UniformUseExistingOutputSubArray:Approx1V2Simple/1.UniformUseReorderedOutputArray:Approx1V2Simple/2.UseExistingOutputSubArray:Approx1V2Simple/2.UniformUseExistingOutputSubArray:Approx1V2Simple/3.UseNullOutputArray:Approx1V2Simple/3.UseFullExistingOutputArray:Approx1V2Simple/3.UseExistingOutputSubArray:Approx1V2Simple/3.UseReorderedOutputArray:Approx1V2Simple/3.UniformUseNullOutputArray:Approx1V2Simple/3.UniformUseFullExistingOutputArray:Approx1V2Simple/3.UniformUseExistingOutputSubArray:Approx1V2Simple/3.UniformUseReorderedOutputArray:Approx1V2Simple/0.UseFullExistingOutputArray:Approx1V2Simple/0.UseExistingOutputSubArray"
```

 - `approx2` passed with gtest:

```
--gtest_filter="-Approx2.CPPNearestBatch:Approx2.CPPLinearBatch:Approx2.CPPNearestMaxDims:Approx2.CPPLinearMaxDims:Approx2.CPPCubicMaxDims:Approx2.OtherDimLinear:Approx2.OtherDimCubic:Approx2Cubic.CPP:Approx2V2Simple/1.UseNullOutputArray:Approx2V2Simple/1.UseFullExistingOutputArray:Approx2V2Simple/1.UseExistingOutputSubArray:Approx2V2Simple/1.UseReorderedOutputArray:Approx2V2Simple/1.UniformUseNullOutputArray:Approx2V2Simple/1.UniformUseFullExistingOutputArray:Approx2V2Simple/1.UniformUseExistingOutputSubArray:Approx2V2Simple/1.UniformUseReorderedOutputArray:Approx2V2Simple/2.UseExistingOutputSubArray:Approx2V2Simple/2.UniformUseExistingOutputSubArray:Approx2V2Simple/3.UseNullOutputArray:Approx2V2Simple/3.UseFullExistingOutputArray:Approx2V2Simple/3.UseExistingOutputSubArray:Approx2V2Simple/3.UseReorderedOutputArray:Approx2V2Simple/3.UniformUseNullOutputArray:Approx2V2Simple/3.UniformUseFullExistingOutputArray:Approx2V2Simple/3.UniformUseExistingOutputSubArray:Approx2V2Simple/3.UniformUseReorderedOutputArray"
```

Checklist
---------
- [X] Rebased on latest master
- [X] Code compiles
- [ ] Tests pass
- [ ] Functions added to unified API
- [ ] Functions documented

### CODE REVIEW

**Review by syurkevi — CHANGES_REQUESTED** [review]

File: src/backend/oneapi/kernel/approx.hpp [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+        h.parallel_for(
+            sycl::nd_range{global, local},
+            approx1Kernel<Ty, Tp>(yoAcc, yo.info, yiAcc, yi.info, xoAcc,
+                                  xo.info, xi_beg, Tp(1) / xi_step, (Ty)offGrid,
+                                  (int)blocksPerMat, (int)batch, (int)method,
+                                  xdim, order));
+    });
+    ONEAPI_DEBUG_FINISH(getQueue());
+}
+
```
Comment (syurkevi) [inline src/backend/oneapi/kernel/approx.hpp:290]: If possible, can we split to two files? approx1 and approx2?

File: src/backend/oneapi/kernel/approx.hpp [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        const int idx = ioff + xid * x_stride;
+
+        for (int n = 0; n < batch; n++) {
+            int idx_n = idx + n * in.strides[batch_dim];
+            d_out[ooff + n * out.strides[batch_dim]] =
+                (doclamp || cond) ? d_in[idx_n] : zero;
+        }
+    }
+
+#if IS_CPLX
```
[the remark this answers is not in the document]
Comment (pv-pterab-s) [reply src/backend/oneapi/kernel/approx.hpp]: I've fixed this up to the point where the compiler complains of types where expected. Tests pass up to expected. 

I'm out of my league on this request, so maybe you can help.

File: src/backend/oneapi/kernel/approx.hpp (hunk 1 above)
Comment (pv-pterab-s) [reply src/backend/oneapi/kernel/approx.hpp]: done

File: src/backend/oneapi/kernel/approx.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (syurkevi) [reply src/backend/oneapi/kernel/approx.hpp]: updated. Interp has now been ripped out and can be reused in other kernels such as _rotate_

**Review by syurkevi — APPROVED** [review]

**Pull request merged**